ros2run/ros2run/api/__init__.py

```diff
@@ -66,6 +67,8 @@ def run_executable(*, path, argv, prefix=None):
             # the subprocess will also receive the signal and should shut down
             # therefore we continue here until the process has finished
             pass
+    if process.returncode < 0:
+        print(signal.strsignal(-process.returncode))
```

Comment:
the alternative is we can wrap subprocess with `shell=True` but i just do not like to instantiate shell process only for this.

Reply:
I don't think we should use `shell=True` either.

As far as the code here, can `returncode` ever be `< 0`?  In Linux, at least, return codes are always 0 or greater.

Reply:
yeah, it is python spec which you can see https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess.returncode.

`returncode` will be negative if the child process is terminated with signal. (e.d `Segmentation Fault` will be `-11`)

Reply:
> yeah, it is python spec which you can see https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess.returncode.

Oh!  I didn't realize that, thanks for pointing it out.

The documentation there says:

```
A negative value -N indicates that the child was terminated by signal N (POSIX only).
```

So I wonder what happens on Windows.

Reply:
that is actually nice question, i am not really sure. (dont have environment.) does anyone happen to know about this?
CC: @iuhilnehc-ynos @Barry-Xu-2018

Reply:
@fujitatomoya  @clalancette 

I check return code on window.  
Run ` ros2 run demo_nodes_cpp talker` and terminate `talker.exe` in windows task manager.  
The return code is `1`.   

According to wiki [exit status for window](https://en.wikipedia.org/wiki/Exit_status#Windows), it provides a reference to [Windows System Error Codes (exit codes)](http://www.hiteksoftware.com/knowledge/articles/049.htm). In this document, exit codes are  all  greater than 0 if error occurs or terminated by other process.

Reply:
thanks @Barry-Xu-2018 

then we can do the following? any suggestions are welcome.

1. check if returcode is NOT zero?
2. if POSIX, call strsignal. and else print general message that subprocess exited with failure.
3. return `returncode`

Reply:
I also did a bit of testing on Windows.  I wrote this silly code:

```
#include <iostream>

int main()
{
    std::cout << "Hello, world" << std::endl;
    *(char *)0 = 0;
    return 0;
}
```

Which obviously causes a crash.  If I compile and run that program on the command-line, I get a negative error code:

```
C:\dev>cl /EHsc crash.cpp
Microsoft (R) C/C++ Optimizing Compiler Version 19.28.29336 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

crash.cpp
Microsoft (R) Incremental Linker Version 14.28.29336.0
Copyright (C) Microsoft Corporation.  All rights reserved.

/out:crash.exe
crash.obj

C:\dev>crash
Hello, world

C:\dev>echo %errorlevel%
-1073741819
```

However, I get a different result if I run it through subprocess on Python.  Here is my simple Python code:

```
import subprocess

process = subprocess.Popen(['crash'])
process.communicate()
print(process.returncode)
```

When I run that, I get:

```
C:\dev>python subprocess-crash.py
Hello, world
3221225477

```

So it looks like Python is wrapping the negative return code or something like that.

Regardless, I think @fujitatomoya 's proposal sounds like an improvement over the current situation, so :+1: from me.